### ISSUE TITLE
Impossible to deserialize a serialized object

### ISSUE THREAD

**Issue by igiona** [tafia/quick-xml#653]
I have a relatively simple XML to serialize and deserialize (a subset of https://en.wikipedia.org/wiki/XML-RPC):
The following code does properly serialize an Object into XML, but the deserialization of the serialized object itself doesn't work.
I'm using:
```toml
[[package]]
name = "quick-xml"
version = "0.30.0"
```

Any hint?

```rust
use serde::{Serialize, Deserialize};

#[derive(Serialize, Deserialize)]
pub struct MethodResponse {
    pub params: Params,
}

#[derive(Serialize, Deserialize)]
pub struct Params {
    pub param: Vec<Param>,
}

#[derive(Serialize, Deserialize, Clone)]
pub struct Param {
    pub value: ValueElement,
}

#[derive(Serialize, Deserialize, Clone)]
pub struct ValueElement {
    pub v: Value,
}

#[derive(Serialize, Deserialize, Clone)]
pub enum Value {
    #[serde(rename = "int")]
    Int(i32),
    #[serde(rename = "double")]
    Double(f64),
    #[serde(rename = "bool")]
    Boolean(bool),
    #[serde(rename = "string")]
    String(String)
}
```

```rust
let obj = MethodResponse 
{ 
    params : Params {
                param : {
                    vec![
                        Param { value: ValueElement { v: Value::String("MyStringValue".to_string()) } },
                        Param { value: ValueElement { v: Value::Boolean(true) } },
                        Param { value: ValueElement { v: Value::Int(66) } },
                        Param { value: ValueElement { v: Value::Double(1.2) } },
                    ]
                }
            }
};

let xml = quick_xml::se::to_string(&obj).unwrap();
println!("{}", xml);
let object: MethodResponse = quick_xml::de::from_str(&xml).unwrap();
```

The serialized XML looks ok to me, and it's correctly printed:

```xml
<MethodResponse><params><param><value><string>Lol</string></value></param><param><value><bool>true</bool></value></param><param><value><int>66</int></value></param><param><value><double>1.2</double></value></param></params></MethodResponse>
```

But the deserialization on the other hand doesn't work and the program panics:
```shell
thread 'xmlrpc::response::tests::deserialize_should_succeed' panicked at 'called `Result::unwrap()` on an `Err` value: Custom("missing field `v`")', src\xmlrpc\response.rs:35:68
stack backtrace:
```

**Comment by Mingun** [comment]
You should add `#[serde(rename = "$value")]` to `pub v: Value`, because deserializer should known to which field it should redirect different tags (`<string>`, `<bool>`, etc.) and that field is a `$value` field. I think, we can emphasise that in the documentation something [here](https://docs.rs/quick-xml/latest/quick_xml/de/index.html).

Also, you should add `#[serde(default)]` to `pub param: Vec<Param>`, otherwise you'll get an error in attempt to deserialize a response with no parameters.

Also, note, that examples in https://docs.rs/quick-xml/latest/quick_xml/de/index.html#value are not correct. They are not tested and I've found that while working on fix for #630. Currently serialization without renames didn't produce `<field/>`, but actual enum variant tags, but deserialization didn't work.

**Comment by igiona** [comment]
> You should add `#[serde(rename = "$value")]` to `pub v: Value`, because deserializer should known to which field it should redirect different tags (`<string>`, `<bool>`, etc.) and that field is a `$value` field. I think, we can emphasise that in the documentation something [here](https://docs.rs/quick-xml/latest/quick_xml/de/index.html).
> 
> Also, you should add `#[serde(default)]` to `pub param: Vec<Param>`, otherwise you'll get an error in attempt to deserialize a response with no parameters.
> 
> Also, note, that examples in https://docs.rs/quick-xml/latest/quick_xml/de/index.html#value are not correct. They are not tested and I've found that while working on fix for #630. Currently serialization without renames didn't produce `<field/>`, but actual enum variant tags, but deserialization didn't work.

Hey @Mingun, many thanks for the fast response! Indeed your suggestions helped 🙏 
Indeed the `$value` feature is not very simple to understand from the documentation, on the other hand probably you kind of assume some "serde" knowledge, which unfortunately I don't yet(?) have. 

Again, many thanks for your support!
